tests/unit/test_umm_var.py

```diff
@@ -758,3 +763,74 @@ def test_export_all_umm_var_to_json(self):
                 saved_json = json.load(file_handler)
 
             self.assertDictEqual(saved_json, umm_var_records[umm_var_index])
+
+
+    @patch('requests.put')
+    def test_publish_umm_var(self, mock_requests_put):
+        ''' Check if `publish_umm_var` returns the expected
+            content for the mocked response.
+        '''
+        # Set the `mock_response` and its status_code
+        mock_response = Mock(spec=requests.Response)
+        mock_response.status_code = 400
+        
+        # Set the return_value of `mock_response` 
+        # And set the return_value of `mock_requests_put` to mock_response
+        mock_response.json.return_value = {'concept-id': 'FOO-EEDTEST'}
+        mock_requests_put.return_value = mock_response
+        umm_var_dict = {'Name': 'Test', 
+                        'MetadataSpecification':
+                            {'URL': 'https://foo.gov/umm/variable/v1.8.2',
+                            'Name': 'UMM-Var',
+                            'Version': '1.8.2'}}
+                
+        concept_id = 'C1256535511-EEDTEST'
+        cmr_env = CMR_UAT
+        token = 'foo'
+        headers_umm_var = {
+            'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
+            + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
+            'Authorization': f'Bearer {token}',
+            'Accept': 'application/json'}
+        
+        url_endpoint = (cmr_env.replace('search', 'ingest') + 'collections/'
+                    f'{concept_id}/variables/{umm_var_dict["Name"]}')
+        
+        response = publish_umm_var(concept_id, umm_var_dict, token, cmr_env)
+        
+        # Check if `publish_umm_var` was called once with expected parameters
+        mock_requests_put.assert_called_once_with(url_endpoint,
+                                                  json=umm_var_dict,
+                                                  headers=headers_umm_var,
+                                                  timeout=10)
+        
+        with self.subTest('Check response when the status code is 400'):
+            self.assertEqual(response.status_code, 400)
+            self.assertEqual(response.json(), {'concept-id': 'FOO-EEDTEST'})
```

Comment:
> Bit of a nit: Nice check for a failed status, but it's a bit weird seeing the JSON response for a successful request with an unsuccessful status.
Yeah, I agree. I think that was the best way I thought for checking if the response failed but I can try tinker with some other ways. 

> Side note - is it worth an extra assertion to know that requests.put was called with the expected JSON content and headers?
I think since we already did the `mock_requests_put.assert_called_once_with()` test we don't need to have it here unless you really thinks it's helpful.

Reply:
I have new commit that addresses this!